From 4fc13270a3b9f9a2d57b7f68ed290de1a4687951 Mon Sep 17 00:00:00 2001
From: chyngyz <chyngyz.omurov@gmail.com>
Date: Wed, 17 Sep 2025 14:50:49 +0600
Subject: [PATCH] Specify linux/amd64 platform for Docker builds

This ensures the Docker images are built for the `linux/amd64` platform, which is necessary for cross-compilation on newer systems, such as M1 Macs.
---
 external-libs/Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/external-libs/Makefile b/external-libs/Makefile
index 96e2b06..da2e90d 100644
--- a/external-libs/Makefile
+++ b/external-libs/Makefile
@@ -8,7 +8,7 @@ mobile: arm64-v8a include/wallet2_api.h VERSION
 arm64-v8a: $(FLAVOUR) android64.Dockerfile
 	-rm -rf arm64-v8a
 	-docker container rm $(FLAVOUR)-android-arm64 -f
-	docker build -f android64.Dockerfile -t $(FLAVOUR)-android-arm64 $(FLAVOUR) --build-arg NPROC=$(NPROC)
+	docker build --platform linux/amd64 -f android64.Dockerfile -t $(FLAVOUR)-android-arm64 $(FLAVOUR) --build-arg NPROC=$(NPROC)
 	docker create -it --name $(FLAVOUR)-android-arm64 $(FLAVOUR)-android-arm64 bash
 	docker cp $(FLAVOUR)-android-arm64:/opt/android/prefix/lib/. arm64-v8a/
 	docker cp $(FLAVOUR)-android-arm64:/src/build/release/lib/. arm64-v8a/$(FLAVOUR)
@@ -16,7 +16,7 @@ arm64-v8a: $(FLAVOUR) android64.Dockerfile
 armeabi-v7a: $(FLAVOUR) android32.Dockerfile
 	-rm -rf armeabi-v7a
 	-docker container rm $(FLAVOUR)-android-arm32 -f
-	docker build -f android32.Dockerfile -t $(FLAVOUR)-android-arm32 $(FLAVOUR) --build-arg NPROC=$(NPROC)
+	docker build --platform linux/amd64 -f android32.Dockerfile -t $(FLAVOUR)-android-arm32 $(FLAVOUR) --build-arg NPROC=$(NPROC)
 	docker create -it --name $(FLAVOUR)-android-arm32 $(FLAVOUR)-android-arm32 bash
 	docker cp $(FLAVOUR)-android-arm32:/opt/android/prefix/lib/. armeabi-v7a
 	docker cp $(FLAVOUR)-android-arm32:/src/build/release/lib/. armeabi-v7a/$(FLAVOUR)
@@ -24,7 +24,7 @@ armeabi-v7a: $(FLAVOUR) android32.Dockerfile
 x86_64: $(FLAVOUR) android64_x86.Dockerfile
 	-rm -rf x86_64
 	-docker container rm $(FLAVOUR)-android-x86_64 -f
-	docker build -f android64_x86.Dockerfile -t $(FLAVOUR)-android-x86_64 $(FLAVOUR) --build-arg NPROC=$(NPROC)
+	docker build --platform linux/amd64 -f android64_x86.Dockerfile -t $(FLAVOUR)-android-x86_64 $(FLAVOUR) --build-arg NPROC=$(NPROC)
 	docker create -it --name $(FLAVOUR)-android-x86_64 $(FLAVOUR)-android-x86_64 bash
 	docker cp $(FLAVOUR)-android-x86_64:/opt/android/prefix/lib/. x86_64
 	docker cp $(FLAVOUR)-android-x86_64:/src/build/release/lib/. x86_64/$(FLAVOUR)
@@ -32,7 +32,7 @@ x86_64: $(FLAVOUR) android64_x86.Dockerfile
 x86: $(FLAVOUR) android32_x86.Dockerfile
 	-rm -rf x86
 	-docker container rm $(FLAVOUR)-android-x86 -f
-	docker build -f android32_x86.Dockerfile -t $(FLAVOUR)-android-x86 $(FLAVOUR) --build-arg NPROC=$(NPROC)
+	docker build --platform linux/amd64 -f android32_x86.Dockerfile -t $(FLAVOUR)-android-x86 $(FLAVOUR) --build-arg NPROC=$(NPROC)
 	docker create -it --name $(FLAVOUR)-android-x86 $(FLAVOUR)-android-x86 bash
 	docker cp $(FLAVOUR)-android-x86:/opt/android/prefix/lib/. x86
 	docker cp $(FLAVOUR)-android-x86:/src/build/release/lib/. x86/$(FLAVOUR)
-- 
2.50.1 (Apple Git-155)

